compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml

```diff
@@ -624,10 +624,25 @@ null : [a] -> Bool
 null [] = True
 null _ = False
 
--- | Filter the list using the function: keep only the elements where the predicate holds.
+-- | Filters the list using the function: keep only the elements where the predicate holds.
 filter : (a -> Bool) -> [a] -> [a]
 filter p = foldr (\x xs -> if p x then x :: xs else xs) []
 
+-- | Filters the list using the applicative function: keeps only the elements where the predicate holds.
+-- Useful for example when a list of contract IDs are known, which should be filtered based on some
+-- condition for the correspodning contracts.
+--
+-- ```
+-- >>> filterA (\x -> Some (x % 2 == 0)) [0..5]
+-- Some [0, 2, 4]
+-- ```
+filterA : Applicative m => (a -> m Bool) -> [a] -> m [a]
+filterA p =
+  foldr filtering (pure [])
+  where
+    filtering a = liftA2 (prependIf a) (p a)
+    prependIf a prepend filtered = if prepend then a :: filtered else filtered
```

Comment:
The implementation looks correct but maybe this one-liner is sweeter?
```suggestion
filterA p = foldr (\ x -> liftA2 (\ flg -> if flg then (x:) else id) (p x)) (pure [])
```

Reply:
In my original implementation I even specified the signatures of the inner functions as well. For me and other colleagues it was really useful to understand the implementation.

If you think this clarity is not needed I'm happy to accept the suggestion.

Reply:
I'm easy but I think we probably favor succinctness over clarity in the context of the standard library.